src/sql/azdata.proposed.d.ts

```diff
@@ -715,4 +715,8 @@ declare module 'azdata' {
 		 */
 		title: string;
 	}
+
+	export namespace workspace {
+		export function createWorkspace(location: vscode.Uri, filePath?: string): Promise<void>;
```

Comment:
why not make the second parameter a Uri and name it workspaceFile to be more clear?

Reply:
When I tried passing it as a Uri, I would get the error "uri.with is not a function" when calling workspaceEditingService.createAndEnterWorkspace(). It worked when I created the Uri in core from the string path, so I think it has something to do with the Uri not getting passed with all the expected attributes from the extension host to core.

Reply:
Keep in mind that when objects cross the JSON RPC boundary while they keep all properties if the object is a class then it won't have the functions such as .with. You need to use URI.revive first on the main thread side in order to recreate it properly before passing it on to stuff that expects an actual URI object

Reply:
ah good to know. Thanks Charles!